tests/Model/CachingIteratorTest.php

```diff
@@ -152,6 +152,12 @@ public function rewind(): void
         $this->assertCount(1, $iterator);
     }
 
+    public function testNonUniqueKeys(): void
```

Comment:
```suggestion
    public function testCountNonUniqueKeys(): void
```

I'd also suggest creating `testIterationNonUniqueKeys()` based on `testIteration()`, which removes `$expectedKey` and does `$this->assertSame(0, $key);` during iteration. Since counting the CachingIterator immediately exhausts the internal iterator, the iteration test would cover the code path of the actual iteration methods (e.g. `next()`, `key()`).